From: Robert Luberda <robert@debian.org>
Date: Sat, 5 Feb 2011 00:36:00 +0100
Subject: 17 close fds

Redirect standard input/output/error to /dev/null and close any other file descriptors.
---
 src/standalone.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/src/standalone.c b/src/standalone.c
index 1d01c00..94faec5 100644
--- a/src/standalone.c
+++ b/src/standalone.c
@@ -120,6 +120,13 @@ int main(int argc, char **argv) {
 	else {
 		standalone = 1;
 	}
+
+	for (tmp = getdtablesize(); tmp >= 0 ; --tmp)
+		close(tmp);
+	if (open("/dev/null", O_RDWR) == STDIN_FILENO) {
+		dup2(STDIN_FILENO, STDOUT_FILENO);
+		dup2(STDIN_FILENO, STDERR_FILENO);
+	}
 #endif
 	pop_openlog();
 	if (atexit(pop_closelog) < 0) {
-- 
